pyrokinetics/gk_code/GKOutputReaderGENE.py

```diff
@@ -340,19 +375,38 @@ def _set_fields(
         # New coords: (field, theta, kx, ky, time)
         fields = fields.transpose(0, 3, 1, 2, 4)
 
-        data["fields"] = (coords, fields)
+        coords = ["theta", "kx", "ky", "time"]
+        for ifield, field_name in enumerate(data["field"].data):
+            field = fields[ifield, :, :, :, :] * gene_field_units[field_name]
+            field = field.to(local_norm.pyrokinetics).magnitude
+            data[field_name] = (coords, field)
+
+        data = data.pint.quantify(pyro_field_units)
 
         return data
 
     @staticmethod
     def _set_fluxes(
-        data: xr.Dataset, raw_data: Dict[str, Any], gk_input: GKInputGENE
+        data: xr.Dataset,
+        raw_data: Dict[str, Any],
+        gk_input: GKInputGENE,
     ) -> xr.Dataset:
         """
         Set flux data over time.
         The flux coordinates should  be (species, moment, field, ky, time)
         """
 
+        local_norm = data.local_norm
+        pyro_flux_units = flux_units(local_norm.pyrokinetics)
+        if gk_input.data["geometry"].get("minor_r", 0.0) == 1.0:
+            gene_flux_units = flux_units(local_norm.pyrokinetics)
+        elif gk_input.data["geometry"].get("major_R", 1.0) == 1.0:
+            gene_flux_units = flux_units(local_norm.gene)
+        else:
+            raise NotImplementedError(
+                "Pyro does handle cases where neither major_R and minor_r are 1.0"
+            )
```

Comment:
```suggestion
            gene_flux_units = flux_units(local_norm.gene)
        elif gk_input.data["geometry"].get("major_R", 1.0) > 1.0:
            gene_flux_units = flux_units(local_norm.pyrokinetics)
        else:
            raise NotImplementedError(
                "Pyro does not handle cases where both major_R is smaller than 1.0 and minor_r is not 1.0"
            )
```
I would suggest this change to force the normalisation to `minor_r` when `major_R > 1`.

Reply:
What is happening here is that we are figuring out what the normalisation is, not setting what pyro stores as the normalisation. 

The normalisation gets forced to a pyro convention at the end here 
https://github.com/pyro-kinetics/pyrokinetics/blob/ae52b411beab0cd205c4a40c2d40adfc28830d16/pyrokinetics/gk_code/GKOutputReaderGENE.py#L475-L478


This is just seeing if we need to re-normalise the output or not. If `minor_r=1` then the run already uses the minor radius as `Lref` so we can leave it as is. If `major_R=1` then the major radius was used and we need to re-normalise. That is what is being done here.

We should probably check for `major_R` first and if that isn't 1 check `minor_r`. If that isn't 1 or isn't defined then we should raise a warning of some kind

Reply:
Of course this change should be reported everywhere in `GKOutputReadearGENE.py` and not just in the above lines.

Reply:
This will cause the program to raise an error but we can put it in the `init_dataset` method so we catch it earlier

Reply:
Yes, I think we can also invert the logic and check `major_R` first, but I think we should check if `major_R>`1 and not if `major_R=1`. 
Maybe it is better in the `init_database` to avoid multiple repetition.